### PR TITLE
[skipruntime-ts] Minor simplifications

### PR DESCRIPTION
**Pull request by mbouaziz**
A few minor simplifications that were waiting on my stack + renaming `sliced` -> `slice`

### CODE REVIEW

File: skipruntime-ts/ts/src/internals/skipruntime_module.ts [hunk 1]
```diff
@@ -628,9 +628,9 @@ class NonEmptyIteratorImpl<T> implements NonEmptyIterator<T> {
     );
 
     return {
-      next() {
+      next(): IteratorResult<T> {
         const value = cloned_iter.next();
-        return { value, done: value == null } as IteratorResult<T>;
+        return value === null ? { value, done: true } : { value };
```
Comment (beauby) [inline skipruntime-ts/ts/src/internals/skipruntime_module.ts:633]: Does it make a difference that `done` is not explicitly set (to `false`) in the else case?

File: skipruntime-ts/ts/src/internals/skipruntime_impl.ts [hunk 2]
```diff
@@ -255,8 +255,7 @@ export class SKStoreImpl extends SkFrozen implements SKStore {
     const name = computeObj.constructor.name;
     const lazyHdl = this.context.lazy(
       name,
-      (selfHdl: LazyCollection<K, V>, key: K) =>
-        computeObj.compute(selfHdl, key),
+      computeObj.compute.bind(computeObj),
```
Comment (beauby) [inline skipruntime-ts/ts/src/internals/skipruntime_impl.ts:258]: Open question: do we generally prefer `obj.method.bind(obj)` over `(arg1, arg2) => obj.method(arg1, arg2)`?

**Review by beauby — APPROVED** [review]
Minor questions, LGTM otherwise.

File: skipruntime-ts/ts/src/internals/skipruntime_module.ts (hunk 1 above)
Comment (mbouaziz) [reply skipruntime-ts/ts/src/internals/skipruntime_module.ts:633]: No, it works as well.

File: skipruntime-ts/ts/src/internals/skipruntime_impl.ts (hunk 2 above)
Comment (mbouaziz) [reply skipruntime-ts/ts/src/internals/skipruntime_impl.ts:258]: With `(arg1, arg2) => obj.method(arg1, arg2)`, I want to simplify to just `obj.method` and then it fails... so `obj.method.bind(obj)`, although ugly, makes it explicit

Comment (beauby) [reply skipruntime-ts/ts/src/internals/skipruntime_impl.ts:258]: In that case let's open an issue with a reminder to use that style across the board (there are instances of the lambda variant in the skipruntime client or server).

**Pull request merged**